packages/workflows/src/handlers/product/revert-update-products.ts

```diff
@@ -0,0 +1,38 @@
+import { Modules, ModulesDefinition } from "@medusajs/modules-sdk"
+import {
+  ProductDTO,
+  ProductTypes,
+  ProductVariantDTO,
+  UpdateProductDTO,
+} from "@medusajs/types"
+
+import { WorkflowArguments } from "../../helper"
+import { UpdateProductsPreparedData } from "./update-products-prepare-data"
+
+type HandlerInput = UpdateProductsPreparedData & {
+  variants: ProductVariantDTO[]
+}
+
+export async function revertUpdateProducts({
```

Comment:
Q: This doesn't cover the full set of updates happening inside product update. The update of product will update all of its relationships (categories, prices and the lot) in the db first, but if one of the following workflows fail, all that was updated in the updateProduct step will not be reverted.

Let me know if this is intentional and is regarded ok for the first step.

Reply:
It should be reverted in this handler which is the last compensation that will be called i.e. here we will re-update (L:30) products with their original data and relationships that are preserved before the workflow has run

Reply:
oh interesting, sounds good. Thanks for pointing it out!